File: 25-04/processos/java/src/pkg/Processo.java
package pkg;

import java.util.Objects;

public class Processo {

    private int id;
    private String descricao;

    /***
     * Construtor do processo
     * @param id - inteiro com número aleatório entre 1 e 1000
     * @param descricao - explicação do processo
     */

    public Processo(int id, String descricao) {
        this.id = id;
        this.descricao = descricao;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    @Override
    public String toString() {
        return "Processo{" +
                "id=" + id +
                ", descricao='" + descricao + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Processo processo = (Processo) o;
        return id == processo.id;
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }
}

File: 25-04/processos/c#/25-04-cs/25-04-cs/Processo.cs
namespace _25_04_cs;

using System;
public class Processo
{
    public int Id { get; set; }
    public string Descricao { get; set; }

    // Construtor do processo
    public Processo(int id, string descricao)
    {
        Id = id;
        Descricao = descricao;
    }

    public override string ToString()
    {
        return $"Processo{{ id={Id}, descricao='{Descricao}' }}";
    }

    public override bool Equals(object obj)
    {
        if (obj == null || GetType() != obj.GetType()) return false;
        Processo processo = (Processo)obj;
        return Id == processo.Id;
    }

    public override int GetHashCode()
    {
        return Id.GetHashCode();
    }
}
